docs/Highlight.md

```diff
@@ -6,115 +6,84 @@ It would be good to support multiple highlighter engines, but also allow for
 ```
 
-
-## Implementation
-
-The summarization/highlight subsystem is implemented using an environment-agnostic
-highlighter/fragmenter, and a higher level which is integrated with RediSearch and
-the query keyword parser.
-
-The summarization process begins by tokenizing the requested field, and splitting
-the document into *fragments*.
-
-When a matching token (or its stemmed variant) is found, a distance counter begins.
-This counts the number of tokens following the matched token. If another matching
-token occurs before the maximum token distance has been exceeded, the counter is
-reset to 0 and the fragment is extended.
-
-Each time a token is found in a fragment, the fragment's score increases. The
-score increase is dependent on the base token score (this is provided as
-input to the fragmenter), and whether this term is being repeated, or if it is
-a new occurrence (within the same fragment). New terms get higher scores; which
-helps eliminate forms like "I said to Abraham: Abraham, why...".
-
-The input score for each term is calculated based on the term's overall frequency
-in the DB (lower frequency means higher score), but this is consider out of bounds
-for the fragmenter.
-
-Once all fragments are scored, they are then *contextualized*. The fragment's
-context is determined to be X amount of tokens surrounding the given matched
-tokens. Words in between the tokens are considered as well, ensuring that every
-fragment is more or less the same size.
\ No newline at end of file
+Highlighting will highlight the found term (and its variants) with a user-defined
+tag. This may be used to display the matched text in a different typeface using
+a markup language, or to otherwise make the text appear differently.
+
+Redis Search can perform highlighting using the `HIGHLIGHT` keyword. If no
+additional arguments are passed, all _returned fields_ are highlighted using
+build-in defaults.
+
+The `HIGHLIGHT` keyword accepts the following arguments:
+
+* **`FIELDS`** If present, must be the first argument. This should be followed
+    by the number of fields to highlight, which itself is followed by a list of
+    fields. Each field present is highlighted. If no `FIELD` directive is passed,
+    then *all* fields returned are highlighted.
+* **`TAGS`** If present, must be followed by two strings; the first is prepended
+    to each term match, and the second is appended to it. If no `TAGS` are
+    specified, a built-in tag value is appended and prepended.
+
+
+#### Field Selection
+
+If no specific fields are passed to the `RETURN`, `SUMMARIZE`, or `HIGHLIGHT`
+keywords, then all of a document's fields are returned. However, if any of these
+keywords contain a `FIELD` directive, then the `SEARCH` command will only retun
+the sum total of all fields enumerated in any of those directives. For example
+in the command `RETURN 1 foo SUMMARIZE FIELDS 1 bar HIGHLIGHT FIELDS 1 baz`,
```

Comment:
That is not the behavior we agreed on.
1. Return always wins. You can't contradict it and return fields that are not mentioned in it.
2. If HIGHLIGHT or SUMMARIZE fields are a superset of RETURN, we ignore anything not in RETURN.
3. highlighted fields and summarized fields can be disjointed but they must obey return.